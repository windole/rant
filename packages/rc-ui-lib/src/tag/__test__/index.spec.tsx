import React from 'react';
import { mount } from 'enzyme';
import { Tag } from '..';

describe('Tag', () => {
  let wrapper;

  afterEach(() => {
    wrapper.unmount();
  });

  it('should emit close event when clicking the close icon', () => {
    const onClose = jest.fn();
    wrapper = mount(<Tag closeable onClose={onClose} />);
    wrapper.find('.rc-tag__close').at(0).simulate('click');
    expect(onClose).toHaveBeenCalledTimes(1);
  });

  it('should hide tag when the show prop is false', () => {
    wrapper = mount(<Tag visible={false} />);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should not trigger click event when clicking the close icon', () => {
    const onClick = jest.fn();
    wrapper = mount(<Tag closeable onClick={onClick} />);
    wrapper.find('.rc-tag__close').at(0).simulate('click');
    expect(onClick).toHaveBeenCalledTimes(0);
    wrapper.simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it('should render border-color correctly', () => {
    wrapper = mount(<Tag plain color="red" textColor="blue" />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
