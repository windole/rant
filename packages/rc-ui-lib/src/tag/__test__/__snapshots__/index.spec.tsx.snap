// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should hide tag when the show prop is false 1`] = `""`;

exports[`Tag should render border-color correctly 1`] = `"<span style=\\"color: blue; border-color: red;\\" class=\\"rc-tag rc-tag--plain rc-tag--default\\"></span>"`;
